src_c/geometry.c

```diff
@@ -7,6 +7,108 @@
 #endif /* ~__AVX2__ */
 
 #define PYGAMEAPI_GEOMETRY_NUMSLOTS 21
+static PyObject *
+pg_raycast(PyObject *self, PyObject *args, PyObject *kwargs)
+{
+    PyObject *startpoint;
+    PyObject *endpoint = NULL;
+    PyObject **farr;
+    PyObject **collisions;
+    Py_ssize_t loop;
+    double angle;
+    double max_dist;
+    double target_pos[2] = {0, 0};
+
+    static char *keywords[] = {"start_pos", "collisions", "endpoint", "angle", "max", NULL};
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "OO|Odd",
+        keywords, &startpoint, &collisions, &endpoint, &angle, &max_dist))
+    {
+        return NULL;
+    }
+
+    farr = PySequence_Fast_ITEMS(collisions);
+
+    if (!PySequence_Check(startpoint)) {
+        return RAISE(PyExc_TypeError, "positions must be sequences");
+    }
+
+    pgLineObject *line = (pgLineObject *)pgLine_Type.tp_new(&pgLine_Type, NULL, NULL);
+    if (line) {
+        line->line.x1 = PyFloat_AsDouble(PySequence_GetItem(startpoint, 0));
+        line->line.y1 = PyFloat_AsDouble(PySequence_GetItem(startpoint, 1));
```

Comment:
This is wrong, PySequence_GetItem is massively inefficient ( we have functions for extracting doubles from PyObject *) and  PyFloat_AsDouble can fail, i see no check here... 
You should use our base functions for this.